src/DotNetty.Buffers/ReadOnlyByteBufferStream.cs

```diff
@@ -29,7 +29,7 @@ public ReadOnlyByteBufferStream(IByteBuffer buffer, bool releaseReferenceOnClosu
 
         public override long Length
         {
-            get { throw new NotSupportedException(); }
+            get => this.buffer.ReadableBytes;
```

Comment:
Ah, of course. The most recent changes I made were to handle/expect the decrease in ReadableBytes, but I didn't put two and two together...

I'll rework this. Thanks!